auto_nag/team_managers.py

```diff
@@ -36,3 +40,81 @@ def get_team_manager(
                 return None
 
         return self.managers[team_name]
+
+    def _fetch_component_teams(self):
+        include_fields = [
+            "components.name",
+            "components.team_name",
+        ]
+
+        def handler(product, data):
+            data.update(
```

Comment:
Nit: `self.component_teams.update` so you don't need the `data` argument and you can avoid passing `product_data`

Reply:
I try to avoid letting the handle functions to access variables out of their local scope when possible. This reduces the complexity since we can think about it as a contained unit. Also, it will be more portable and can be moved without modification.

The mentioned benefits do not apply in the is case since the handler is pretty simple. But I prefer keeping it as is for consistency.